code/lib/core-events/src/errors/server-errors.ts

```diff
@@ -394,6 +395,99 @@ export class NoMatchingExportError extends StorybookError {
   }
 }
 
+export class MainFileESMOnlyImportError extends StorybookError {
+  readonly category = Category.CORE_SERVER;
+
+  readonly code = 5;
+
+  public documentation =
+    'https://github.com/storybookjs/storybook/issues/23972#issuecomment-1948534058';
+
+  constructor(
+    public data: { location: string; line: string | undefined; num: number | undefined }
+  ) {
+    super();
+  }
+
+  template() {
+    const message = [
+      `Storybook failed to load ${this.data.location}..`,
+      '',
+      `It looks like the file tried to load/import an ESM only module.`,
+      `Support for this is currently limited in ${this.data.location}.`,
+      `You can import ESM modules in your main file, but only as dynamic import.`,
+      '',
+    ];
+    if (this.data.line) {
+      message.push(
+        white(
+          `In your ${yellow(this.data.location)} file, line ${bold.cyan(
+            this.data.num
+          )} threw an error, which looks like this:`
```

Comment:
I read the message as if you want to show what the error looks like, but you actually want to show the line, right?
```suggestion
          )} threw an error:`
```